packages/signers/src/lit-protocol/signer.ts

```diff
@@ -0,0 +1,200 @@
+            pkpPublicKey: `0x${this._pkpPublicKey}`,
+            expiration: params.expiration,
+            resources: params.resources,
+            chainId: 1,
+          });
+          return response.authSig;
+        };
+      } else {
+        authNeededCallback = async (params: AuthCallbackParams) => {
+          const response = await this._client.signSessionKey({
+            statement: params.statement,
+            sessionKey: sessionKeypair,
+            authMethods: [props.context as LitAuthMethod],
+            pkpPublicKey: `0x${this._pkpPublicKey}`,
+            expiration: params.expiration,
+            resources: params.resources,
+            chainId: 1,
+          });
+          return response.authSig;
+        };
+      }
+
+      if (!this._client.ready) {
+        await this._client.connect();
+      }
+
+      const sessionSigs = await this._client
+        .getSessionSigs({
+          chain,
+          expiration:
+            props.expiration ??
+            new Date(Date.now() + 1000 * 60 * 60 * 24 * 7).toISOString(),
+          resourceAbilityRequests: resourceAbilities,
+          authNeededCallback,
+        })
+        .catch((err) => {
+          console.log(
+            "error while attempting to access session signatures: ",
+            err
+          );
+          throw err;
+        });
+
+      this._authContext = props.context;
+      this._session = sessionSigs;
+
+      this._signer = new PKPEthersWallet({
+        pkpPubKey: this._pkpPublicKey,
+        rpc: this._rpcUrl,
+        controllerSessionSigs: sessionSigs as LitSessionSigsMap,
+      });
+
+      await this._signer.init();
+      this.inner = this._signer;
+
+      this._authContext = sessionSigs as C;
+    } else {
+      this._authContext = props.context;
+      this._session = props.context as SessionSigsMap;
+
+      this._signer = new PKPEthersWallet({
+        pkpPubKey: this._pkpPublicKey,
+        rpc: this._rpcUrl,
+        controllerSessionSigs: this._authContext as LitSessionSigsMap,
+      });
+
+      await this._signer.connect();
+      this.inner = this._signer;
+    }
+
+    return this._session;
+  };
+
+  getAuthDetails = async (): Promise<LitSessionSigsMap> => {
+    this._checkInternals();
+    return this._authContext as LitSessionSigsMap;
+  };
+
+  getAddress = async () => {
+    this._checkInternals();
+    const address = await this.inner?.getAddress();
+
+    return address as `0x${string}`;
+  };
+
+  signMessage = async (msg: Uint8Array | string) => {
+    this._checkInternals();
+
+    return this.inner?.signMessage(msg) as Promise<`0x${string}`>;
+  };
+
+  signTypedData = (params: SignTypedDataParams) => {
+    this._checkInternals();
+
+    return this.inner?._signTypedData(
+      params.domain as TypedDataDomain,
+      params.types as any,
+      params.message
+    ) as Promise<`0x${string}`>;
+  };
+
+  private _checkInternals() {
```

Comment:
this method was created to prevent duplicating the same checks in every function. the purpose was not to remove a null check, as we still want to throw an explicit exception.